### PR TITLE
Doc improvements

### PR DESCRIPTION
**Pull request by jonas**
As discussed on Gitter a few minor tweaks to the Scaladoc and document limitation regarding parsing of numerical values on Scala.js.

### CODE REVIEW

File: docs/src/main/tut/parsing.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ position: 1
 
 # Parsing JSON
 
-Circe includes a parsing module, which is a wrapper around the [Jawn][jawn] JSON parser.
+Circe includes a parsing module, which is on the JVM is a wrapper around the [Jawn][jawn] JSON parser and for JavaScript uses the built-in [`JSON.parse`][json.parse].
```
Comment (travisbrown) [inline docs/src/main/tut/parsing.md]: Nitpick: "which ~~is~~ on the JVM is".

File: docs/src/main/tut/parsing.md [hunk 2]
```diff
@@ -60,4 +60,16 @@ import cats.syntax.either._
 val json: Json = parse(rawJson).getOrElse(Json.Null)
 ```
 
+## Warnings and known issues
+
+For Scala.js, the result of parsing JSON is restricted to the expressiveness of
+the JavaScript types. In particular numerical values may [loose precision][#393]
```
Comment (travisbrown) [inline docs/src/main/tut/parsing.md]: s/loose/lose/

File: docs/src/main/tut/parsing.md [hunk 3]
```diff
@@ -60,4 +60,16 @@ import cats.syntax.either._
...
+when decoded by [`JSON.parse`][json.parse] to a JavaScript number. For example:
```
Comment (travisbrown) [inline docs/src/main/tut/parsing.md]: I'm not sure "JavaScript number" is exactly the right phrasing here—it seems more like the problem is that you're trying to decode to a Scala type like `Long` that has its own semantics, even if it's implemented using JavaScript numbers on Scala.js. I'm happy to work on details like this in a follow-up PR, though.

File: .gitignore [hunk 4]
```diff
@@ -8,3 +8,4 @@ target/
 .project
 .classpath
 tmp/
+docs/src/main/tut/contributing.md
```
Comment (travisbrown) [inline .gitignore]: Thanks—I keep meaning to open a PR for this but hadn't yet.

File: docs/src/main/tut/parsing.md (hunk 1 above)
Comment (jonas) [inline docs/src/main/tut/parsing.md]: Thanks

File: docs/src/main/tut/parsing.md (hunk 3 above)
Comment (jonas) [inline docs/src/main/tut/parsing.md]: OK, I've tried to clarify further. On Scala.js `Long`s are represented with [high/low pair](https://www.scala-js.org/api/scalajs-library/latest/#scala.scalajs.runtime.RuntimeLong) to ensure the Scala semantics and `StringLike.toLong` and `java.lang.long.parseLong` works correctly for the issue mentioned in #393. My understanding is that JSON.parse coerces the value into a double at which point then circe's JSON AST converter cannot do anything about it.

**Comment by travisbrown** [comment]
:+1:

**Pull request merged**